### PR TITLE
PR 2 of 4 for SNV Consensus: Merge callers script

### PR DESCRIPTION
**Pull request by cansavvy**
### Purpose/implementation

This is a side analysis that originated from [this original PR](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/174) but after some reorganization has been split up and spread out. 

This second PR merges the callers' VAF and TMB files for an output of four files. These four files can optionally be used in a later script to make plots and/or be used by a later script to make the final mutation consensus call. 

**The four output files:**
- `all_callers_vaf.rds` - contains all the VAF file information for all callers.
- `all_callers_tmb.rds` - contains all the TMB file information for all callers.
- `mutation_id_list.rds` - a full list of the mutations that can be used for an UpSetR graph
- `callers_per_mutation.rds` - contains a breakdown for each mutation of what callers called it. Will be used to identify the consensus mutations.

### Issue

For SNV caller comparison #161, #103 and Tumor Mutation Burden #3 and sort of #11

### Directions for reviewers

I don't like having to use `suppressWarnings` to have it ignore combining columns. Is there a more seamless way to make this section less "shoot-in-the-foot" like? 
Some of the writing steps are quite memory intensive and will undoubtedly add to the run time of the CircleCI tests. Do you have suggestions for decreasing the run time for this script?

### Results

This script doesn't add results, but is a the next step toward getting the snv consensus. 

#### Docker and continuous integration

These items were taken care of in #183
- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile. 
- [x] This analysis has been added to continuous integration.

#### PR Checklist
* [x] Run a linter
* [x] Set the seed- I didn't, but let me know if I should.
* [x] Comments and/or documentation up to date 
* [x] Double check your paths
* [x] Spell check any Rmd file or md file
* [x] Restart R and run all notebooks fresh and save

### CODE REVIEW

File: analyses/snv-callers/scripts/03-merge_callers.R [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+#              FALSE
+#
+#
+# Command line example:
+#
+# Rscript 03-merge_callers.R \
+# -v results \
+# -o strelka2 \
+# -s wxs \
+# -w
```
Comment (jaclyn-taroni) [inline analyses/snv-callers/scripts/03-merge_callers.R]: What does this `-w` correspond to?

Comment (cansavvy) [reply analyses/snv-callers/scripts/03-merge_callers.R]: I used to have `--overwrite` have a short flag but then I deleted it. Forgot to delete it from the example.

File: analyses/snv-callers/scripts/03-merge_callers.R [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+)
+
+# Print this out to check
+message("Merging these TMB files: \n", paste0(tmb_files, "\n"))
+
+################################### Set Up #####################################
+# Set and make the plots directory
+opt$output <- file.path(root_dir, opt$output)
+
+# Make caller specific plots folder
```
Comment (jaclyn-taroni) [inline analyses/snv-callers/scripts/03-merge_callers.R]: Update this comment

File: analyses/snv-callers/scripts/03-merge_callers.R [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+  })
+
+  # Carry over the callers' names
+  names(vaf_list) <- caller_names
+
+  # Print progress message
+  message("Saving master VAF file to: \n", all_vaf_file)
+
+  # Combine and save VAF file
+  vaf_df <- suppressWarnings(dplyr::bind_rows(vaf_list, .id = "caller")) %>%
```
Comment (jaclyn-taroni) [inline analyses/snv-callers/scripts/03-merge_callers.R]: Can you put a detailed comment here explaining why you're using `suppressWarnings` and why the warnings are expected?

File: analyses/snv-callers/scripts/03-merge_callers.R [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+  warning("Option used for file format (-f) is not supported. Only 'tsv' or 'rds'
+          files are supported. Defaulting to rds.")
+  opt$file_format <- "rds"
+  file_suffix <- "rds"
+}
+
+# Normalize this file path
+opt$vaf <- file.path(root_dir, opt$vaf)
+
+# Check the output directory exists
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply analyses/snv-callers/scripts/03-merge_callers.R]: Are you asking if `opt$vaf` is designating the input directory? Yes it is.

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/snv-callers/scripts/03-merge_callers.R]: Yes, but look at this comment.

[the remark this answers is not in the document]
Comment (jaclyn-taroni) [reply analyses/snv-callers/scripts/03-merge_callers.R]: The comment says `# Check the output directory exists` - this is the input directory for this script.

**Pull request merged**